targets/tofino/base/expression_resolver.cpp

```diff
@@ -23,7 +24,636 @@ const IR::Expression *TofinoBaseExpressionResolver::processTable(const IR::P4Tab
+     /// Load the counter with an immediate value or a header field.
+     {"ParserCounter.set",
+      {"value"},
+      [](const ExternMethodImpls::ExternInfo &externInfo) {
+          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
+                            externInfo.externObjectRef.toString(), externInfo.methodName);
+          return nullptr;
+      }},
+     /// Load the counter with a header field.
+     /// @param max : Maximum permitted value for counter (pre rotate/mask/add).
+     /// @param rotate : Right rotate (circular) the source field by this number of bits.
+     /// @param mask : Mask the rotated source field by 2 ^ (mask + 1) - 1.
+     /// @param add : Constant to add to the rotated and masked lookup field.
+     {"ParserCounter.set",
+      {"field", "max", "rotate", "mask", "add"},
+      [](const ExternMethodImpls::ExternInfo &externInfo) {
+          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
+                            externInfo.externObjectRef.toString(), externInfo.methodName);
+          return nullptr;
+      }},
+     /// @return true if counter value is zero.
+     {"ParserCounter.is_zero",
+      {},
+      [](const ExternMethodImpls::ExternInfo &externInfo) {
+          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
+                            externInfo.externObjectRef.toString(), externInfo.methodName);
+          return nullptr;
+      }},
+     /// @return true if counter value is negative.
+     {"ParserCounter.is_negative",
+      {},
+      [](const ExternMethodImpls::ExternInfo &externInfo) {
+          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
+                            externInfo.externObjectRef.toString(), externInfo.methodName);
+          return nullptr;
+      }},
+     /// Add an immediate value to the parser counter.
+     /// @param value : Constant to add to the counter.
+     {"ParserCounter.increment",
+      {"value"},
+      [](const ExternMethodImpls::ExternInfo &externInfo) {
+          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
+                            externInfo.externObjectRef.toString(), externInfo.methodName);
+          return nullptr;
+      }},
+     /// Subtract an immediate value from the parser counter.
+     /// @param value : Constant to subtract from the counter.
+     {"ParserCounter.decrement",
+      {"value"},
+      [](const ExternMethodImpls::ExternInfo &externInfo) {
+          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
+                            externInfo.externObjectRef.toString(), externInfo.methodName);
+          return nullptr;
+      }},
+
+     // -----------------------------------------------------------------------------
+     // PARSER PRIORITY
+     // -----------------------------------------------------------------------------
+     // Tofino ingress parser compare the priority with a configurable!!! threshold
```

Comment:
Hmm, anything we can do here? I think because this only hits in drop cases there is not much optimization possible.

Reply:
Yes. I will implement other externs including this one in the next pr.